presto-raptorx/src/main/java/com/facebook/presto/raptorx/RaptorConnector.java

```diff
@@ -0,0 +1,178 @@
+    private final TransactionWriter transactionWriter;
+    private final ConnectorSplitManager splitManager;
+    private final ConnectorPageSourceProvider pageSourceProvider;
+    private final ConnectorPageSinkProvider pageSinkProvider;
+    private final ConnectorNodePartitioningProvider nodePartitioningProvider;
+    private final List<PropertyMetadata<?>> sessionProperties;
+    private final List<PropertyMetadata<?>> tableProperties;
+    private final Set<Procedure> procedures;
+    private final Set<SystemTable> systemTables;
+
+    @Inject
+    public RaptorConnector(
+            LifeCycleManager lifeCycleManager,
+            TransactionManager transactionManager,
+            NodeSupplier nodeSupplier,
+            BucketManager bucketManager,
+            TransactionWriter transactionWriter,
+            ConnectorSplitManager splitManager,
+            ConnectorPageSourceProvider pageSourceProvider,
+            ConnectorPageSinkProvider pageSinkProvider,
+            ConnectorNodePartitioningProvider nodePartitioningProvider,
+            RaptorSessionProperties sessionProperties,
+            RaptorTableProperties tableProperties,
+            Set<Procedure> procedures,
+            Set<SystemTable> systemTables)
+    {
+        this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.nodeSupplier = requireNonNull(nodeSupplier, "nodeSupplier is null");
+        this.bucketManager = requireNonNull(bucketManager, "bucketManager is null");
+        this.transactionWriter = requireNonNull(transactionWriter, "transactionWriter is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.pageSinkProvider = requireNonNull(pageSinkProvider, "pageSinkProvider is null");
+        this.nodePartitioningProvider = requireNonNull(nodePartitioningProvider, "nodePartitioningProvider is null");
+        this.sessionProperties = requireNonNull(sessionProperties, "sessionProperties is null").getSessionProperties();
+        this.tableProperties = requireNonNull(tableProperties, "tableProperties is null").getTableProperties();
+        this.procedures = ImmutableSet.copyOf(requireNonNull(procedures, "procedures is null"));
+        this.systemTables = ImmutableSet.copyOf(requireNonNull(systemTables, "systemTables is null"));
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+    {
+        checkConnectorSupports(REPEATABLE_READ, isolationLevel);
+        return transactionManager.create();
+    }
+
+    @Override
+    public void commit(ConnectorTransactionHandle transactionHandle)
+    {
+        Transaction transaction = transactionManager.remove(transactionHandle);
```

Comment:
Sorry I need to write more comments. It's a complex project and I can't finish in one round. 

After line 104, there is a low chance issue:

transactionManager remove(t1);
CommitCleaner.java line 144,  abortTransactions(transactionManager.activeTransactions(), so t1 is aborted;
DatabasemetadataWriter line 276, tries to commit t1, but writing into "transactions" fails.

I did observe this in raptor-dev

Reply:
Good catch! Fixed